flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/utils/JobResultUtils.java

```diff
@@ -32,4 +37,23 @@ public static void assertSuccess(final JobResult result) {
 			}
 		}
 	}
+
+	public static void assertIncomplete(final CompletableFuture<JobResult> result) {
+		if (result.isDone()) {
+			final JobResult jobResult;
+			try {
+				jobResult = result.get();
+			} catch (InterruptedException | ExecutionException e) {
+				// we already know it is done so this doesn't happen
+				throw new AssertionError("Unexpected exception when processing finished future.", e);
+			}
+
+			if (jobResult.isSuccess()) {
+				throw new AssertionError("Job finished successfully.");
+			} else {
+				// this is guarantee to fail with an error
+				assertSuccess(jobResult);
```

Comment:
How about to extract the error throwing block in `assertSuccess` into a reusable method?
So that we can use it here to to avoid the confusing assertion?

Reply:
yeah we could do that.

Reply:
Addedd `JobResultUtils#throwAssertionErrorOnFailedResult` for this purpose in another commit f523e291218ebbcd66632cb88c962ea22a2b3355.